libs/ledger/src/protocols/main_chain_rpc_service.cpp

```diff
@@ -457,9 +459,22 @@ MainChainRpcService::State MainChainRpcService::OnSynchronised(State current, St
 
   FETCH_UNUSED(current);
 
-  if (chain_.HasMissingBlocks())
+  MainChain::BlockPtr head_of_chain = chain_.GetHeaviestBlock();
+  uint64_t const      seconds_since_last_block =
+      GetTime(fetch::moment::GetClock("default", fetch::moment::ClockType::SYSTEM)) -
+      head_of_chain->timestamp;
+
+  // Assume if the chain is quite old that there are peers who have a heavier chain we haven't
+  // heard about
+  if (seconds_since_last_block > 100 && head_of_chain->block_number != 0)
```

Comment:
this avoids the case where over half the nodes crash, and start to re-sync but stall since they think they are synced as there are no loose blocks. There are no loose blocks because the miners at the head of the chain are waiting for them to catch up so they can generate entropy.

Reply:
This seems like something that needs making a variable / constant